### PR TITLE
UNR-2129 Removed NetDriver from WorkerConnection

### PR DESCRIPTION
**Pull request by aleximprobable**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
Removed NetDriver from WorkerConnection.

#### Release note
No release note update.

#### Tests
Everything should work as before.

#### Documentation
No doc update.

#### Primary reviewers
@samiwh @m-samiec

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/SpatialWorkerConnection.h [hunk 1]
```diff
@@ -83,20 +89,21 @@ class SPATIALGDK_API USpatialWorkerConnection : public UObject, public FRunnable
 	UGlobalStateManager* GlobalStateManager;
 
 private:
-	void ConnectToReceptionist(bool bConnectAsClient);
-	void ConnectToLocator();
+	void ConnectToReceptionist(bool bConnectAsClient, uint32 PlayInEditorID);
+	void ConnectToLocator();
```
Comment (samiwh) [inline SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/SpatialWorkerConnection.h]: nit - erroneous whitespace

File: SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/SpatialWorkerConnection.h [hunk 2]
```diff
@@ -83,20 +89,21 @@ class SPATIALGDK_API USpatialWorkerConnection : public UObject, public FRunnable
 	UGlobalStateManager* GlobalStateManager;
 
 private:
-	void ConnectToReceptionist(bool bConnectAsClient);
-	void ConnectToLocator();
+	void ConnectToReceptionist(bool bConnectAsClient, uint32 PlayInEditorID);
```
Comment (samiwh) [inline SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/SpatialWorkerConnection.h]: nit - ID is normally Id I think. It's special.

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 3]
```diff
@@ -786,7 +800,7 @@ void USpatialNetDriver::BeginDestroy()
 			{
 				Cast<USpatialGameInstance>(LocalWorld->GetGameInstance())->DestroySpatialWorkerConnection();
 			}
-
+			Connection->UnbindCallbacks();
```
Comment (samiwh) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: This is a bit weird. Can it not just unbind it's own callbacks.

Comment (samiwh) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: Or can this not be part of the instance destroying the connection above?

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 4]
```diff
@@ -355,6 +358,17 @@ void USpatialNetDriver::OnConnectedToSpatialOS()
 	}
 }
 
+void USpatialNetDriver::OnConnectionToSpatialOSFailed(uint8_t ConnectionStatusCode, const FString ErrorMessage)
```
Comment (samiwh) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]:
```suggestion
void USpatialNetDriver::OnConnectionToSpatialOSFailed(uint8_t ConnectionStatusCode, const FString& ErrorMessage)
```

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 5]
```diff
@@ -32,6 +32,7 @@
 #include "SpatialConstants.h"
 #include "SpatialGDKSettings.h"
 #include "Utils/EntityPool.h"
+#include "Utils/ErrorCodeRemapping.h"
```
Comment (samiwh) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: What is this used for?

File: SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/SpatialWorkerConnection.h (hunk 1 above)
Comment (aleximprobable) [reply SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/SpatialWorkerConnection.h]: Removed.

File: SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/SpatialWorkerConnection.h (hunk 2 above)
Comment (aleximprobable) [reply SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/SpatialWorkerConnection.h]: Discussed. NetDriver has it as `ID`, so keeping it like that for consistency.

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp (hunk 3 above)
Comment (aleximprobable) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: Removed.

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp (hunk 4 above)
Comment (aleximprobable) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: Updated.

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp (hunk 5 above)
Comment (aleximprobable) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: It's used for `ENetworkFailure::FromDisconnectOpStatusCode(ConnectionStatusCode)`

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 6]
```diff
@@ -355,6 +358,17 @@ void USpatialNetDriver::OnConnectedToSpatialOS()
 	}
 }
 
+void USpatialNetDriver::OnConnectionToSpatialOSFailed(uint8_t ConnectionStatusCode, const FString& ErrorMessage)
+{
+	if (const USpatialGameInstance * GameInstance = GetGameInstance())
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]:
```suggestion
	if (const USpatialGameInstance* GameInstance = GetGameInstance())
```

**Pull request merged**